### DISCUSSION TITLE
Can't get YAML files to be read as global data

### DISCUSSION THREAD

**Discussion by nityasampath** [11ty/eleventy#2656] [Q&A] [answered]
I am trying to set up eleventy to use YAML for global data. I find it much easier to type out than something like json. I followed the instructions in the documentation, but it's not working. My .eleventy.js config file looks like this.

```
const yaml = require("js-yaml");
const inspect = require("util").inspect;

module.exports = config => {

  config.addDataExtension("yml, yaml", (contents) => yaml.safeLoad(contents));

  // Set directories to pass through to the _site folder
  config.addPassthroughCopy('./assets/images/');

  config.addFilter("debug", (content) => `<pre>${inspect(content)}</pre>`);

  return {
    markdownTemplateEngine: 'njk',
    dataTemplateEngine: 'njk',
    htmlTemplateEngine: 'njk',
    dir: {
      includes: '_includes',
      layouts: '_layouts',
    }
  };
};
```

I have been trying to set up my site navigation using a YAML data file, which looks like this

```
- name: Home
  link: /
- name: About
  link: /about.html
- name: Projects
  link: /projects.html
- name: Contact
  link: /contact.html
```

But it is not working. I tried setting up a debugging filter to output all the global data, so I could see if the YAML data was being compiled and it's not. It doesn't even show up in the global data. 

I did try writing the data in json and was actually able to get the contents to show up as global data. But I would really like to get it to work as YAML and I don't know why it won't.

I don't know what I am missing. Does anyone have any advice to get this working?

**Comment by pdehaan** [comment, marked as answer]
Re: https://www.11ty.dev/docs/data-custom/#usage

> `Coming soon in v2.0.0-canary.10` Pass a comma-separated list of extensions.

Are you using v2 canary branch? Or the stable v1.0.2 branch?

**Comment by amithm7** [comment]
```js
  eleventyConfig.addDataExtension("yaml", (contents) => yaml.load(contents));
```

this works for me.
